docs/versioned_docs/version-v0.3/guides/sign-in-with-github-google-facebook-linkedin.mdx

```diff
@@ -99,6 +99,64 @@ selfservice:
 Next, open the login endpoint of the SecureApp and you should see the GitHub
 Login option!
 
+## Microsoft
+
+This will enable you to log in using any Azure AD directory - Multitenant and personal Microsoft accounts (e.g. Skype, Xbox) depending on the settings made when creating the application in Azure AD.
+
+To set up "Sign in with Microsoft" you must first register an application
+[Register an application with the Microsoft identity platform](https://docs.microsoft.com/en-us/azure/active-directory/develop/quickstart-register-app).
+
+It might be nessecary to enable the implicit grant flow for access tokens and ID tokens.
+
+Set the "Authorization callback URL" to:
+
+```
+https://public.server.fqdn.example.org/.ory/kratos/public/self-service/browser/flows/strategies/oidc/callback/azuread
+```
+
+The pattern of this URL is:
+
+```
+http(s)://<domain-of-ory-kratos>:<public-port>/self-service/browser/flows/strategies/oidc/callback/<provider-id>
+```
+
+```json title="contrib/quickstart/kratos/email-password/oidc.azuread.jsonnet"
+# claims contains all the data sent by the upstream.
+local claims = std.extVar('claims');
+{
+  identity: {
+    traits: {
+      email: claims.email
```

Comment:
Does AzureAD include `email_verified`? If so, we should probably add a warning here as we did for Google/GitHub: https://github.com/ory/kratos/pull/433/files#diff-2a62fee703dc56d70e23792e010a5a01L67-L73

Reply:
There is an optional claim, verified_primary_email that can be queried. 
https://docs.microsoft.com/en-us/azure/active-directory/develop/active-directory-optional-claims

Reply:
@aeneasr - I read up on the issue on the Microsoft documentation, it seems like the same issue with unverified e-mails applies for Microsoft accounts as for GitHub, however, if you are using a corporate AzureAD (one you have control over), this does not apply and the email is to be considered as "verified". I did try to extract the verified_primary_email token that should theoretically work for this, however I did not have success at the moment (and it is not needed for my use-case). I think the documentation I submitted should be good to go with the warning / links attached.

Reply:
Ok, sounds good!